python-package/xgboost/spark/model.py

```diff
@@ -222,11 +216,11 @@ def saveImpl(self, path):
         """
         xgb_model = self.instance._xgb_sklearn_model
         _SparkXGBSharedReadWrite.saveMetadata(self.instance, path, self.sc, self.logger)
-        model_save_path = os.path.join(path, "model.json")
-        ser_xgb_model = serialize_xgb_model(xgb_model)
-        _get_spark_session().createDataFrame(
-            [(ser_xgb_model,)], ["xgb_sklearn_model"]
-        ).write.parquet(model_save_path)
+        model_save_path = os.path.join(path, "model")
+        xgb_model_file = save_model_to_json_file(xgb_model)
+        # The json file written by Spark base on `booster.save_raw("json").decode("utf-8")`
+        # can't be loaded by XGBoost directly.
+        _get_spark_session().read.text(xgb_model_file).write.text(model_save_path)
```

Comment:
```
_get_spark_session().read.text(xgb_model_file).
```
This line is not correct.
spark.read.text(path) the path must be a distributed file system path which all spark executor can access.

Reply:
You can use distributed FS API to copy local file `xgb_model_file` into the model saved path (a hadoop FS path)

Reply:
wow, right. you're correct, @WeichenXu123 Good findings. Could you point me to what is the "distributed FS API"? Really appreciate it.

Reply:
You can use this:
https://arrow.apache.org/docs/python/generated/pyarrow.fs.HadoopFileSystem.html

But, this does not support DBFS (databricks filesystem), we need support databricks case as well.
Databricks mount `dbfs:/xxx/xxx` to local file system `/dbfs/xxx/xxx`.

Reply:
The example code in the PR description 
```
import xgboost as xgb
bst = xgb.Booster()

# Basically, YOUR_MODEL_PATH should be like "xxxx/model/xxx.txt"
YOUR_MODEL_PATH="xxx"
bst.load_model(YOUR_MODEL_PATH)
```
seems does not wok ? If the path is a distributed FS path ?

Reply:
@WeichenXu123, I use the RDD to save the text file, it should work with all kinds of hadoop-compatible FS..